tests/test_models_dbi_utils.py

```diff
@@ -42,13 +42,13 @@ def test_select_best_dbr_generator_and_run(backend, nqubits, step):
     initial_off_diagonal_norm = dbi.off_diagonal_norm
 
     for _ in range(NSTEPS):
-        idx, step_optimize = select_best_dbr_generator_and_run(
+        idx, step_optimize, flip_sign = select_best_dbr_generator_and_run(
             dbi, Z_ops, step=step, compare_canonical=True
         )
         if idx == len(Z_ops):
             dbi(step=step_optimize, mode=DoubleBracketGeneratorType.canonical)
             dbi.mode = DoubleBracketGeneratorType.single_commutator
         else:
-            dbi(step=step_optimize, d=Z_ops[idx])
+            dbi(step=step_optimize, d=flip_sign * Z_ops[idx])
```

Comment:
@andrea-pasquale @MatteoRobbiati @Edoardo-Pedicillo 

This is a key change which implements for the strategy of using $D=Z_1^{idx_0}\ldots Z_L^{idx_L}$ the theoretical discussion roughly saing for DBI either +D or -D will be the right one. Here Sam implemented it by checking the CS angle.

In general, we can now keep an eye on this for other strategies and basically check if $\langle W_{test}, W_{canonical}\rangle>0$, otherwise flip. 

Thanks @Sam-XiaoyueLi